### ISSUE TITLE
Add support for setting driver-class-name property on datasource driver

### ISSUE THREAD

**Issue by bjwschaap** [biemond/biemond-wildfly#82]
Right now the module only supports the `driver-xa-datasource-class-name` property on a JDBC driver. However, when using a non-XA driver, the `driver-class-name` property could be more appropriate.

A direction might be to add a driver_class_name parameter to wildfly::datasources::driver, differentiate between a XA and non-XA driver, or make wildfly::datasources::driver 'transparent' by using e.g. a hash instead of named parameters/arguments.

**Comment by jairojunior** [comment]
You're right, I just did the change, but be aware that with JDBC4+ you have SPI for java.sql.Driver, hence you do not need to explicitly specify driver-class-name.

Thank you for your time.

**Comment by bjwschaap** [comment]
Unfortunately it fails:
```
Error: Could not set 'present' on ensure: Failed with: WFLYJCA0034: Unable to instantiate driver class "undef". See log (WARN) for more details for {"address":[{"subsystem":"datasources"},{"jdbc-driver":"postgresqlXa"}],"operation":"add","driver-name":"postgresqlXa","driver-module-name":"org.postgresql","driver-class-name":"undef","driver-xa-datasource-class-name":"org.postgresql.xa.PGXADataSource"} at 16:/modules/wildfly/manifests/util/resource.pp
Error: Could not set 'present' on ensure: Failed with: WFLYJCA0034: Unable to instantiate driver class "undef". See log (WARN) for more details for {"address":[{"subsystem":"datasources"},{"jdbc-driver":"postgresqlXa"}],"operation":"add","driver-name":"postgresqlXa","driver-module-name":"org.postgresql","driver-class-name":"undef","driver-xa-datasource-class-name":"org.postgresql.xa.PGXADataSource"} at 16:/modules/wildfly/manifests/util/resource.pp
Wrapped exception:
Failed with: WFLYJCA0034: Unable to instantiate driver class "undef". See log (WARN) for more details for {"address":[{"subsystem":"datasources"},{"jdbc-driver":"postgresqlXa"}],"operation":"add","driver-name":"postgresqlXa","driver-module-name":"org.postgresql","driver-class-name":"undef","driver-xa-datasource-class-name":"org.postgresql.xa.PGXADataSource"}
Error: /Stage[main]/Wildfly_wrapper::Config/Wildfly::Datasources::Driver[postgresql-9.2-XA]/Wildfly::Util::Resource[/subsystem=datasources/jdbc-driver=postgresqlXa]/Wildfly_resource[/subsystem=datasources/jdbc-driver=postgresqlXa]/ensure: change from absent to present failed: Could not set 'present' on ensure: Failed with: WFLYJCA0034: Unable to instantiate driver class "undef". See log (WARN) for more details for {"address":[{"subsystem":"datasources"},{"jdbc-driver":"postgresqlXa"}],"operation":"add","driver-name":"postgresqlXa","driver-module-name":"org.postgresql","driver-class-name":"undef","driver-xa-datasource-class-name":"org.postgresql.xa.PGXADataSource"} at 16:/modules/wildfly/manifests/util/resource.pp
Error: Could not set 'present' on ensure: Failed with: WFLYJCA0034: Unable to instantiate driver class "undef". See log (WARN) for more details for {"address":[{"subsystem":"datasources"},{"jdbc-driver":"db2"}],"operation":"add","driver-name":"db2","driver-module-name":"com.ibm.db2","driver-class-name":"undef","driver-xa-datasource-class-name":"com.ibm.db2.jcc.DB2Driver"} at 16:/modules/wildfly/manifests/util/resource.pp
```
Parameters that are `undef` shouldn't be passed to `wildfly::util::resource`, or filtered in the provider.

**Comment by bjwschaap** [comment]
Created pull request #84 for a suggested fix.

**Comment by jairojunior** [comment]
`wildfly_resource` should be inspecting resource state hash for undef values. I'll deeply investigate later what's happening.

In the first versions I was using stdlib `delete_undef_values`function on wildfly::util::resource::state, but not sure why it was dropped.

**Comment by jairojunior** [comment]
Just fixed in PR #85. Not sure why, but for a long time my I left delete_undef_values($content) out of this repository.

**Comment by bjwschaap** [comment]
Hi @jairojunior thanx, problem is fixed now.
